src/IonParserTextRaw.ts

```diff
@@ -1072,13 +1074,9 @@ export class ParserTextRaw {
                         }
                     } else if (this.isLowSurrogate(ch)) {
                         throw new Error("unexpected low surrogate: " + ch);
-                    } else if (t === T_STRING3 && ch === CH_SQ) {
-                        if (this.verifyTriple(index)) {
+                    } else if (t === T_STRING3 && ch === CH_SQ && !bs && this.verifyTriple(index)) {
                             index = this._skip_triple_quote_gap(index, this._end, /*acceptComments*/ true);
-                        } else {
-                            s += String.fromCharCode(ch);
-                        }
-                    } else {
+                    } else if (ch >= 0) {
```

Comment:
this character check skips characters like escaped newlines which return negative values.